group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorResult.java

```diff
@@ -80,10 +86,28 @@ public CoordinatorResult(
         List<U> records,
         T response,
         CompletableFuture<Void> appendFuture
+    ) {
+        this(records, response, appendFuture, appendFuture == null);
+    }
+
+    /**
+     * Constructs a Result with records, a response, an append-future, and replayRecords.
+     *
+     * @param records
+     * @param response
+     * @param appendFuture
+     * @param replayRecords
+     */
+    public CoordinatorResult(
+        List<U> records,
+        T response,
+        CompletableFuture<Void> appendFuture,
+        boolean replayRecords
     ) {
         this.records = Objects.requireNonNull(records);
         this.response = response;
         this.appendFuture = appendFuture;
+        this.replayRecords = Objects.requireNonNull(replayRecords);
```

Comment:
`Objects.requireNonNull` does not seem necessary here as `replayRecords` is a primitive type.

Reply:
It's because `replayRecord` is a `Boolean` instead of `boolean`. And this is because we need to use `replayRecord.hashCode()` to compute the hash for the CoordinatorResult.